src/twisted/web/http_headers.py

```diff
@@ -209,23 +218,50 @@ def setRawHeaders(self, name, values):
                 "instance of %r instead" % (name, type(values))
             )
 
+        if not isinstance(name, (bytes, str)):
+            raise TypeError(
+                "Header name is an instance of %r, " "not bytes or str" % (type(name),)
+            )
+
+        for count, value in enumerate(values):
+            if not isinstance(value, (bytes, str)):
+                raise TypeError(
+                    "Header value at position %s is an instance of %r, not "
+                    "bytes or str"
+                    % (
+                        count,
+                        type(value),
+                    )
+                )
+
```

Comment:
Why is this included in this PR? It's not adding type info; it's doing something else.

Reply:
I guess from a backward compat POV, this isn't really a problem, but it's not clear that calling isinstance on each item is worth the work; I mean it wasn't before, and now with the type checking, you should be catching such things with `mypy`, no?

Reply:
When `_sanitizeLinearWhitespace` was introduced it added an implicit type assertion (technically this was a backwards-incompatible change; it caught a number of places where non-string values were passed in some codebases I'm aware of, fortunately all in tests). When this happens it produces a mildly cryptic exception, so I'm in favor of the much-better exceptions added here.

Reply:
This came from the original PR done by @hawkowl here: https://github.com/twisted/twisted/pull/1284/files#diff-effe5888987605608f990d71470cc65dR229

Reply:
ok